packages/grid/x-data-grid-pro/src/components/GridRowReorderCell.tsx

```diff
@@ -85,7 +89,11 @@ const GridRowReorderCell = (params: GridRenderCellParams) => {
   };
 
   return (
-    <div className={classes.root} draggable={isDraggable} {...draggableEventHandlers}>
+    <div
+      className={classes.root}
+      draggable={isDraggable}
+      {...(isDraggable && draggableEventHandlers)}
```

Comment:
Maybe move `isDraggable` check to the `draggableEventHandlers` declaration to avoid creating event handlers on every render if they won't be used anyway?

```js
const draggableEventHandlers = isDraggable ? {
  onDragStart: publish('rowDragStart'),
  onDragOver: publish('rowDragOver'),
  onDragEnd: publish('rowDragEnd'),
} : null;
```